Test flake in TestCheckpointer/UnscheduleCheckpointer

https://jenkins-kube-lifecycle.prod.coreos.systems/job/bootkube-pr-e2e-calico/243/console

```
[1094973.894078] golang[6]: === RUN   TestCheckpointer
[1094973.894593] golang[6]: === RUN   TestCheckpointer/UnscheduleCheckpointer
[1095623.597129] golang[6]: === RUN   TestCheckpointer/UnscheduleParent
[1095923.635633] golang[6]: --- FAIL: TestCheckpointer (949.74s)
[1095923.645399] golang[6]:     --- FAIL: TestCheckpointer/UnscheduleCheckpointer (649.70s)
[1095923.645627] golang[6]:     	checkpointer_test.go:131: Failed to verifyCheckpoint: should not find checkpoint "test-checkpointer" in "/etc/kubernetes/checkpoint-secrets/bootkube-e2e-58f3f1da-unschedulecheckpointer", error: <nil>, output: "test-checkpointer-m9vl9"
```

--- a/e2e/checkpointer_test.go
+++ b/e2e/checkpointer_test.go
@@ -127,9 +127,14 @@ func testCheckpointerUnscheduleCheckpointer(t *testing.T) {
 	if err := verifyPod(c, "test-nginx", true); err != nil {
 		t.Fatalf("Failed to verifyPod: %s", err)
 	}
-	if err := verifyCheckpoint(c, testNS, "test-checkpointer", false, false); err != nil {
-		t.Fatalf("Failed to verifyCheckpoint: %s", err)
-	}
+
+	// Currently disabled due to e2e testing flakes. See:
+	// * https://github.com/kubernetes-incubator/bootkube/issues/816
+	// * https://github.com/kubernetes-incubator/bootkube/pull/836
+	// if err := verifyCheckpoint(c, testNS, "test-checkpointer", false, false); err != nil {
+	//   t.Fatalf("Failed to verifyCheckpoint: %s", err)
+	// }
+
 	if err := verifyCheckpoint(c, testNS, "test-nginx", false, false); err != nil {
 		t.Fatalf("Failed to verifyCheckpoint: %s", err)
 	}

